cdap-watchdog/src/main/java/co/cask/cdap/logging/pipeline/kafka/KafkaLogProcessorPipeline.java

```diff
@@ -604,4 +588,75 @@ public String toString() {
         '}';
     }
   }
+
+  /**
+   * Transforms kafka {@code MessageAndOffset} iterator to {@code ProcessorEvent} iterator.
+   */
+  private final class KafkaMessageTransformIterator implements Iterator<ProcessorEvent<KafkaOffset>> {
+    private final String topic;
+    private final int partition;
+    private final Iterator<MessageAndOffset> kafkaMessageIterator;
+    private ProcessorEvent<KafkaOffset> nextEntry;
+    private MessageAndOffset nextMessage;
+
+    KafkaMessageTransformIterator(String topic, int partition, Iterator<MessageAndOffset> kafkaMessageIterator) {
+      this.topic = topic;
+      this.partition = partition;
+      this.kafkaMessageIterator = kafkaMessageIterator;
+    }
+
+    @Override
+    public boolean hasNext() {
+      if (nextEntry != null) {
+        return true;
+      }
+
+      if (!kafkaMessageIterator.hasNext()) {
+        return false;
+      }
+
+      boolean skipped;
+      do {
```

Comment:
this is needed because we want to skip events that throw `IOException`. However, this should not happen in normal scenario, unless some garbage is published to logs kafka topic.